include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -891,6 +900,9 @@ struct OrtApi {
   ORT_API2_STATUS(AddSessionConfigEntry, _Inout_ OrtSessionOptions* options,
                   _In_z_ const char* config_key, _In_z_ const char* config_value);
 
+  // Set the languate projection for collecting telemetry data before creating session
+  ORT_API2_STATUS(SetLanguageProjection, _In_ const OrtEnv* ort_env, _In_ OrtLanguageProjection projection);
```

Comment:
Do you anticipate more stuff to be added to telemetry using the public C API? If yes, I think it would be more convenient for you to expose an opaque telemetry object, set things in this object using a generic string key-value pair (similar to AddSessionConfigEntry) and then finally add this object to OrtEnv. This way you don't have to add new telemetry related APIs to multiple bindings.

Reply:
As talked offline.  we don't have near plan to add more stuff

---
In reply to: [482338858](https://github.com/microsoft/onnxruntime/pull/5023#discussion_r482338858) [](ancestors = 482338858)